src/main/java/org/cascadebot/cascadebot/commands/music/SkipCommand.java

```diff
@@ -0,0 +1,125 @@
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        AudioTrack track = context.getMusicPlayer().getPlayer().getPlayingTrack();
+        if(track == null) {
+            context.getTypedMessaging().replyDanger("I'm not playing anything!");
+            return;
+        }
+
+        if(context.getMusicPlayer().getTracks().peek() == null) {
+            context.getTypedMessaging().replyDanger("No song to skip to!");
+            return;
+        }
+
+        if(context.getArgs().length > 0) {
+            if (context.getArg(0).equalsIgnoreCase("force")) {
+                if(context.hasPermission("skip.force")) {
+                    context.getMusicPlayer().skip();
+                    context.getTypedMessaging().replySuccess("Forcefully skipped the song");
+                } else {
+                    context.getUIMessaging().sendPermissionError("skip.force");
+                }
+                return;
+            }
+        }
+
+        if(voteMap.containsKey(context.getGuild().getIdLong())) {
+            if(context.getArgs().length > 0) {
+                if(context.getArg(0).equalsIgnoreCase("yes")) {
+                    voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.TICK);
+                    return;
+                } else if(context.getArg(0).equalsIgnoreCase("no")) {
+                    voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.RED_CROSS);
+                    return;
+                }
+            }
+            voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.TICK);
+            context.getTypedMessaging().replyWarning("A skip vote is already running, but we added your vote automatically");
+            return;
+        }
+
+        if(track.getUserData().equals(sender.getUser().getIdLong())) {
+            context.getMusicPlayer().skip();
+            context.getTypedMessaging().replySuccess("Skipped currently playing song because you queued it");
+            return;
+        }
+
+        VoteButtonGroupBuilder buttonGroupBuilder = new VoteButtonGroupBuilder(VoteMessageType.YES_NO);
+        buttonGroupBuilder.addExtraButton(new Button.UnicodeButton(UnicodeConstants.FAST_FORWARD, (runner, channel, message) -> {
+            //TODO check perm
+            message.delete().queue();
+            voteMap.get(context.getGuild().getIdLong()).stopVote();
+            voteMap.remove(context.getGuild().getIdLong());
+            context.getMusicPlayer().skip();
+            context.getTypedMessaging().replySuccess("Forcefully skipped the song");
+        }));
+        buttonGroupBuilder.setPeriodicConsumer((results, message) -> {
+            StringBuilder resultsBuilder = new StringBuilder();
+            for(VoteResult result : results) {
+                resultsBuilder.append(result.getVote()).append(" - ").append(result.getAmount()).append(' ');
```

Comment:
This should append a newline also

Reply:
I prefer how it looks as is.

Reply:
👍 - 1 👎 - 2

Is how it would look? Looks ok tbh 🤔

Reply:
yegh that. except with the tick and cross emotes